parasolr/query.py

```diff
@@ -161,67 +167,52 @@ def get_facets(self) -> Dict[str, int]:
 
     @staticmethod
     def _lookup_to_filter(key: str, value: Any) -> str:
-        """Convert keyword/value argument, with optional filters, into a
-        Solr query. Currently supports simple key-value pairs, searching for
-        empty/unset values, and __in filters of lists.
+        """Convert keyword/value argument, with optional lookups separated by
+        ``__``, including: in and exists. Field names should *NOT* include
+        double-underscores by convention.
 
             Returns: A propertly formatted Solr query string.
         """
-
-        # Flag if a null/empty search is being conducted
-        null_filter = False
-
-        # Base filter for all queries
-        _filter = ''
-
+        # NOTE: Should this go to a class or module constant?
+        ALL_VALUES = '[* TO *]'
+        # check for a lookup separator and split
+        lookup = ''
+        split_key = key.split(LOOKUP_SEP)
+        if len(split_key) == 1:
+            # simple lookup, return key,value pair
+            return '%s:%s' % (key, value)
         # Implementations of Django-style filters such as __in=[a, b, c]
-        # or __range=(start, end).
-
-        # Parse these special cases first and handle simple key/value pair
-        # at the very end.
-
-        # Regex to split for a double-underscore at end of a filter
-        # since Solr fields, in theory, could have double-underscores in name
-        double_underscore_regex = re.compile(r'_{2}(?=[A-Za-z]+$)')
-
-        # value is a list, join with OR logic for all values in list,
-        # include the empty value for a missing/empty field
-        if key.endswith('__in'):
-            # filter out a negative search and set flag
-            null_filter = bool([item for item in value if not item])
-
-            value = [item for item in value if item]
-            key = re.split(double_underscore_regex, key)[0]
+        # or __range=(start, end)
+
+        # NOTE: Assuming there is only one LOOKUP_SEP without error handling
+        key, lookup = split_key
+
+        if lookup == 'in':
+            # value is a list, join with OR logic for all values in list,
+            # treat '' or None values as flagging an exists query
+            implicit_exists = False
```

Comment:
`implicit_exists` seems like a confusing name for this. Isn't sort of an explicit not exists?

I think for consistency with the `exists` filter, what about just `not_exists` ?